File: scripts/capture_promo_screenshots.py
#!/usr/bin/env python3
"""Capture real NeerCred screenshots — minimal API calls, full workflow."""

from pathlib import Path
import time

import requests

OUT = Path("/opt/cursor/artifacts/neercred-promo-video/screenshots")
BASE = "http://localhost:3000"
API = "http://localhost:8000/api"


def api_session(mobile: str) -> str:
    time.sleep(1)
    r = requests.post(f"{API}/auth/send-otp", json={"mobile": mobile, "sms_consent": True})
    r.raise_for_status()
    otp = r.json().get("dev_otp", "123456")
    r = requests.post(f"{API}/auth/verify-otp", json={"mobile": mobile, "otp": otp})
    r.raise_for_status()
    return r.json()["session_token"]


def complete_profile(token: str) -> dict:
    profile = {
        "session_token": token,
        "pan": "ABCDE1234F",
        "full_name": "Rahul Sharma",
        "date_of_birth": "1990-05-15",
        "email": "rahul.sharma@email.com",
        "pincode": "411001",
        "gender": "male",
        "monthly_income": 75000,
        "employment_type": "salaried",
        "city": "Pune",
        "consents": {
            "dpdp_data_processing": True,
            "privacy_policy": True,
            "terms_of_service": True,
            "credit_bureau_check": True,
        },
    }
    requests.post(f"{API}/leads/details", json=profile).raise_for_status()
    requests.post(f"{API}/leads/check-eligibility", json={
        "session_token": token, "loan_purpose": "wedding", "existing_emi": 0,
    }).raise_for_status()
    return requests.get(f"{API}/leads/offers", params={"session_token": token}).json()


def select_internal_offer(token: str, offers: dict) -> int | None:
    if not offers.get("offers"):
        return None
    internal = [o for o in offers["offers"] if o.get("workflow_mode") != "external_handoff"]
    o = internal[0] if internal else offers["offers"][0]
    r = requests.post(f"{API}/leads/select-offer", json={
        "session_token": token,
        "offer_id": o["offer_id"],
        "lender_name": o["lender_name"],
        "loan_amount": o["loan_amount"],
        "interest_rate": o["interest_rate"],
        "tenure_months": o["tenure_months"],
        "emi": o["emi"],
        "lender_data_sharing_consent": True,
    })
    if r.ok:
        return r.json().get("application_id")
    print("  ! select:", r.status_code, r.text[:150])
    return None


def run_capture():
    from playwright.sync_api import sync_playwright

    OUT.mkdir(parents=True, exist_ok=True)
    app_id = None
    offers = {"offers": []}

    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        ctx = browser.new_context(viewport={"width": 390, "height": 844}, device_scale_factor=2, is_mobile=True)
        page = ctx.new_page()

        def shot(name):
            page.wait_for_timeout(1200)
            page.screenshot(path=str(OUT / f"{name}.png"))
            print(f"  ✓ {name}")

        def dismiss():
            for sel in ['button:has-text("Accept all")', 'button:has-text("Accept")']:
                try:
                    if page.locator(sel).first.is_visible(timeout=1000):
                        page.locator(sel).first.click()
                        page.wait_for_timeout(300)
                except Exception:
                    pass

        def set_token(token: str | None):
            page.evaluate(
                f"() => {{ {'localStorage.setItem(' + repr('session_token') + ', ' + repr(token) + ')' if token else 'localStorage.removeItem(' + repr('session_token') + ')'} }}"
            )

        # 01 Homepage
        page.goto(BASE, wait_until="networkidle", timeout=60000)
        dismiss()
        shot("01-homepage")

        # 02-05 Profile flow via browser (one OTP)
        set_token(None)
        page.goto(f"{BASE}/apply", wait_until="networkidle", timeout=60000)
        dismiss()
        shot("02-mobile")

        page.locator('input[type="tel"]').first.fill("9123456789")
        page.locator('input[type="checkbox"]').first.check()
        page.locator('button:has-text("Continue")').first.click()
        page.wait_for_timeout(2000)
        shot("03-otp")

        # Read dev OTP from page
        otp = "123456"
        try:
            otp_text = page.locator("strong.font-mono").first.text_content(timeout=3000)
            if otp_text and len(otp_text.strip()) == 6:
                otp = otp_text.strip()
        except Exception:
            pass
        page.locator('input[inputmode="numeric"], input[placeholder*="•"]').first.fill(otp)
        page.locator('button:has-text("Verify")').first.click()
        page.wait_for_timeout(2500)
        shot("04-profile-pan")

        page.locator('input[maxlength="10"], input[placeholder*="ABCDE" i]').first.fill("ABCDE1234F")
        page.wait_for_timeout(2500)
        page.locator('button:has-text("Continue")').first.click()
        page.wait_for_timeout(2000)
        shot("05-profile-details")

        # 06-07 Offers + KYC via API (single session)
        token = api_session("9988776655")
        offers = complete_profile(token)
        set_token(token)
        page.goto(f"{BASE}/apply", wait_until="networkidle", timeout=60000)
        dismiss()
        try:
            page.locator('text=/Your offers|eligible/i').first.wait_for(timeout=10000)
        except Exception:
            pass
        shot("06-offers")
        page.evaluate("window.scrollTo(0, 500)")
        page.wait_for_timeout(800)
        shot("06b-offer-cards")

        app_id = select_internal_offer(token, offers)
        if app_id:
            set_token(token)
            page.goto(f"{BASE}/application/{app_id}/kyc", wait_until="networkidle", timeout=60000)
            dismiss()
            shot("07-kyc")

        page.goto(f"{BASE}/dashboard?demo=1", wait_until="networkidle", timeout=60000)
        shot("08-dashboard")
        page.goto(f"{BASE}/compliance", wait_until="networkidle", timeout=60000)
        shot("10-compliance")

        browser.close()

    print(f"\nDone. Offers: {len(offers.get('offers', []))}, KYC: {app_id}")


if __name__ == "__main__":
    run_capture()

File: scripts/capture_promo_screenshots.js
#!/usr/bin/env node
/**
 * Capture real NeerCred mobile screenshots for promo video.
 * Run: node scripts/capture_promo_screenshots.js
 */
const { chromium } = require("playwright");
const fs = require("fs");
const path = require("path");

const OUT = "/opt/cursor/artifacts/neercred-promo-video/screenshots";
const BASE = "http://localhost:3000";
const MOBILE = { width: 390, height: 844 };

async function shot(page, name) {
  const file = path.join(OUT, `${name}.png`);
  await page.screenshot({ path: file, fullPage: false });
  console.log("  ✓", name);
  return file;
}

async function dismissOverlays(page) {
  try {
    const cookie = page.locator('button:has-text("Accept")');
    if (await cookie.isVisible({ timeout: 2000 })) await cookie.click();
  } catch {}
  try {
    const close = page.locator('[aria-label="Close"]');
    if (await close.isVisible({ timeout: 1000 })) await close.click();
  } catch {}
}

async function main() {
  fs.mkdirSync(OUT, { recursive: true });
  const browser = await chromium.launch({ headless: true });
  const ctx = await browser.newContext({
    viewport: MOBILE,
    deviceScaleFactor: 2,
    isMobile: true,
    userAgent:
      "Mozilla/5.0 (iPhone; CPU iPhone OS 17_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.0 Mobile/15E148 Safari/604.1",
  });
  const page = await ctx.newPage();

  // 01 Homepage
  await page.goto(BASE, { waitUntil: "networkidle", timeout: 60000 });
  await dismissOverlays(page);
  await page.waitForTimeout(1500);
  await shot(page, "01-homepage");

  // 02 Apply - Mobile
  await page.goto(`${BASE}/apply`, { waitUntil: "networkidle", timeout: 60000 });
  await dismissOverlays(page);
  await page.waitForTimeout(1000);
  await shot(page, "02-mobile");

  // 03 OTP - fill mobile
  const mobileInput = page.locator('input[type="tel"], input[placeholder*="mobile" i], input[placeholder*="digit" i]').first();
  await mobileInput.fill("9876543210");
  try {
    const consent = page.locator('input[type="checkbox"]').first();
    if (await consent.isVisible({ timeout: 2000 })) await consent.check();
  } catch {}
  const continueBtn = page.locator('button:has-text("Continue"), button:has-text("continue")').first();
  await continueBtn.click();
  await page.waitForTimeout(2000);
  await shot(page, "03-otp");

  // 04 OTP verify - read dev OTP from page
  let otp = "123456";
  try {
    const devOtpEl = page.locator('text=/Dev mode OTP/i');
    if (await devOtpEl.isVisible({ timeout: 3000 })) {
      const text = await page.locator('strong.font-mono').first().textContent();
      if (text && text.length === 6) otp = text.trim();
    }
  } catch {}
  const otpInputs = page.locator('input[maxlength="1"], input[inputmode="numeric"]');
  const count = await otpInputs.count();
  if (count >= 6) {
    for (let i = 0; i < 6; i++) await otpInputs.nth(i).fill(otp[i]);
  } else {
    const single = page.locator('input[placeholder*="OTP" i], input[placeholder*="•" i]').first();
    if (await single.isVisible({ timeout: 2000 })) await single.fill(otp);
  }
  await page.waitForTimeout(500);
  const verifyBtn = page.locator('button:has-text("Verify"), button:has-text("verify")').first();
  await verifyBtn.click({ force: false });
  await page.waitForTimeout(3000);
  await shot(page, "04-profile-pan");

  // 05 Profile - fill and advance through all sub-steps
  async function fillCurrentField() {
    const textInput = page.locator('input[type="text"], input[type="email"], input[type="number"], input[type="date"]').first();
    if (await textInput.isVisible({ timeout: 800 }).catch(() => false)) {
      const ph = ((await textInput.getAttribute("placeholder")) || "").toLowerCase();
      if (ph.includes("pan")) await textInput.fill("ABCDE1234F");
      else if (ph.includes("name")) await textInput.fill("Rahul Sharma");
      else if (ph.includes("email")) await textInput.fill("rahul@email.com");
      else if (ph.includes("pin")) await textInput.fill("411001");
      else if (ph.includes("income")) await textInput.fill("75000");
      else if (ph.includes("city")) await textInput.fill("Pune");
      else if (ph.includes("emi")) await textInput.fill("0");
      else await textInput.fill("Test");
    }
    const radios = page.locator('input[type="radio"]');
    if (await radios.first().isVisible({ timeout: 500 }).catch(() => false)) {
      await radios.first().check().catch(() => {});
    }
    const selects = page.locator("select");
    if (await selects.first().isVisible({ timeout: 500 }).catch(() => false)) {
      await selects.first().selectOption({ index: 1 }).catch(() => {});
    }
    const checkboxes = page.locator('input[type="checkbox"]');
    const cbCount = await checkboxes.count();
    for (let j = 0; j < cbCount; j++) {
      try { await checkboxes.nth(j).check({ timeout: 300 }); } catch {}
    }
  }

  for (let step = 0; step < 14; step++) {
    const offersVisible = await page.locator('h2:has-text("offer"), text=/Your offers/i').first()
      .isVisible({ timeout: 800 }).catch(() => false);
    if (offersVisible) break;
    await fillCurrentField();
    const btn = page.locator('button:has-text("Continue"), button:has-text("See my offers")').first();
    if (!(await btn.isVisible({ timeout: 2000 }).catch(() => false))) break;
    const disabled = await btn.isDisabled().catch(() => true);
    if (disabled) {
      await fillCurrentField();
      await page.waitForTimeout(500);
    }
    if (await btn.isDisabled().catch(() => true)) break;
    await btn.click();
    await page.waitForTimeout(1200);
    if (step === 2) await shot(page, "05-profile-details");
    if (step === 6) await shot(page, "05b-consent");
  }
  await shot(page, "06-offers");

  // 07 Dashboard demo
  await page.goto(`${BASE}/dashboard?demo=1`, { waitUntil: "networkidle", timeout: 60000 });
  await page.waitForTimeout(1500);
  await shot(page, "07-dashboard");

  // 08 Track
  await page.goto(`${BASE}/track`, { waitUntil: "networkidle", timeout: 60000 });
  await page.waitForTimeout(1000);
  await shot(page, "08-track");

  // 09 Compliance
  await page.goto(`${BASE}/compliance`, { waitUntil: "networkidle", timeout: 60000 });
  await page.waitForTimeout(1000);
  await shot(page, "09-compliance");

  // 10 Rates
  await page.goto(`${BASE}/rates`, { waitUntil: "networkidle", timeout: 60000 });
  await page.waitForTimeout(1000);
  await shot(page, "10-rates");

  await browser.close();
  console.log("\nScreenshots saved to", OUT);
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});
